src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Compositor/DevicePortalConnection.cs

```diff
@@ -0,0 +1,310 @@
+        {
+            async Task Connect()
+            {
+                var info = await SendJSONRequest<InfoResponse>(InfoRoute, "GET", connectCTS.Token);
+                DeviceName = info.ComputerName;
+
+                var packages = await SendJSONRequest<PackagesResponse>(InstalledPackagesRoute);
+                var optDeviceApp = packages.InstalledPackages?.FirstOrDefault(p => p.Name.Contains(HolographicCameraAppName));
+                if (optDeviceApp?.PackageFullName == null)
+                {
+                    Debug.LogError("Could not find holographic camera app installed on device");
+                }
+                else
+                {
+                    deviceApp = optDeviceApp.Value;
+                }
+
+                await RunHealthCheck();
+            }
+
+            State = DevicePortalState.Connecting;
+            this.ipAddress = ipAddress;
+            // mind the "auto-" prefix to bypass CSRF protection
+            authorization = "Basic " + Convert.ToBase64String(Encoding.UTF8.GetBytes("auto-" + user + ":" + password));
+            connectCTS = new CancellationTokenSource();
+            Dispatcher.ScheduleAsync(Connect, connectCTS.Token);
+        }
+
+        public void CancelConnect()
+        {
+            if (connectCTS != null)
+            {
+                connectCTS.Cancel();
+                State = DevicePortalState.NotConnected;
+            }
+        }
+
+        public void Disconnect()
+        {
+            State = DevicePortalState.NotConnected;
+        }
+
+        public void StartApp()
+        {
+            async Task StartAppAsync()
+            {
+                var request = await SendRequest(StartAppRoute, "POST");
+                if (request.isHttpError || request.isNetworkError)
+                {
+                    Debug.LogError("Could not send a request to start the app: " + request.error);
+                    State = DevicePortalState.NotStarted;
+                    return;
+                }
+
+                for (int i = 0; i < appStartMaxTries; i++)
+                {
+                    await CheckRunningProcess();
+                    if (State == DevicePortalState.Running)
+                        return;
+                    await Task.Delay(TimeSpan.FromSeconds(appStartCheckInterval));
+                }
+
+                Debug.LogError("Could not start the app");
+            }
+
+            if (!IsAppInstalled)
+            {
+                Debug.LogError("App cannot be started because it is not installed");
+                return;
+            }
+            if (State != DevicePortalState.NotStarted)
+            {
+                Debug.LogError($"Unexpected state for starting the app: {State}");
+                return;
+            }
+            State = DevicePortalState.Starting;
+            var cts = new CancellationTokenSource();
+            Dispatcher.ScheduleAsync(StartAppAsync, cts.Token);
```

Comment:
nit: if we aren't ever cancelling these CancellationTokenSources declared in line, it may be more legible to use CancellationToken.None instead